src/components/application_manager/test/commands/mobile/show_constant_tbt_test.cc

```diff
@@ -0,0 +1,381 @@
+
+    smart_objects::SmartObject msg_params =
+        smart_objects::SmartObject(smart_objects::SmartType_Map);
+
+    msg_params[am::strings::turn_icon][am::strings::value] = kTutnIconValue;
+    msg_params[am::strings::next_turn_icon][am::strings::value] =
+        kNextTutnIconValue;
+    msg_params[am::strings::navigation_text_1] = kNavigationText1;
+    msg_params[am::strings::navigation_text_2] = kNavigationText2;
+    msg_params[am::strings::eta] = kEta;
+    msg_params[am::strings::total_distance] = kTotalDistance;
+    msg_params[am::strings::time_to_destination] = kTimeToDestination;
+    msg_params[am::strings::soft_buttons] = "sds";
+
+    (*msg)[am::strings::msg_params] = msg_params;
+
+    return msg;
+  }
+
+  MessageSharedPtr CreateExpectedMessage() {
+    MessageSharedPtr msg = CreateMessage(smart_objects::SmartType_Map);
+
+    smart_objects::SmartObject msg_params =
+        smart_objects::SmartObject(smart_objects::SmartType_Map);
+
+    int32_t index = 0;
+
+    msg_params[am::hmi_request::navi_texts][index]
+              [am::hmi_request::field_name] = static_cast<int32_t>(
+                  hmi_apis::Common_TextFieldName::navigationText1);
+    msg_params[am::hmi_request::navi_texts][index++]
+              [am::hmi_request::field_text] = kNavigationText1;
+
+    msg_params[am::hmi_request::navi_texts][index]
+              [am::hmi_request::field_name] = static_cast<int32_t>(
+                  hmi_apis::Common_TextFieldName::navigationText2);
+    msg_params[am::hmi_request::navi_texts][index++]
+              [am::hmi_request::field_text] = kNavigationText2;
+
+    msg_params[am::hmi_request::navi_texts][index]
+              [am::hmi_request::field_name] =
+                  static_cast<int32_t>(hmi_apis::Common_TextFieldName::ETA);
+    msg_params[am::hmi_request::navi_texts][index++]
+              [am::hmi_request::field_text] = kEta;
+
+    msg_params
+        [am::hmi_request::navi_texts][index][am::hmi_request::field_name] =
+            static_cast<int32_t>(hmi_apis::Common_TextFieldName::totalDistance);
+    msg_params[am::hmi_request::navi_texts][index++]
+              [am::hmi_request::field_text] = kTotalDistance;
+
+    msg_params[am::hmi_request::navi_texts][index]
+              [am::hmi_request::field_name] = static_cast<int32_t>(
+                  hmi_apis::Common_TextFieldName::timeToDestination);
+    msg_params[am::hmi_request::navi_texts][index++]
+              [am::hmi_request::field_text] = kTimeToDestination;
+
+    (*msg)[am::strings::msg_params] = msg_params;
+
+    return msg;
+  }
+
+  void ExpectedManageMobileCommandResultCode(mobile_apis::Result::eType code) {
+    EXPECT_CALL(app_mngr_, ManageMobileCommand(MobileResultCodeIs(code), _))
+        .WillOnce(Return(true));
+  }
+
+ private:
+  void SetUp() OVERRIDE {
+    message_ = CreateFullParamsUISO();
+    request_ = CreateCommand<ShowConstantTBTRequest>(message_);
```

Comment:
@BSolonenko suggest to create command in every test case explicitly, because some of your tests contain `message_` modification

Reply:
@AKalinich-Luxoft Fixed in https://github.com/smartdevicelink/sdl_core/pull/1977/commits/2127b6028b54933a3f410a26f186212ea4845467